VoterWeightPlugins/hooks/useCalculatedVoterWeight.ts

```diff
@@ -0,0 +1,34 @@
+import { ProgramAccount, TokenOwnerRecord } from "@solana/spl-governance";
+import {CalculatedWeight, UseVoterWeightPluginsArgs, VoterWeightPluginInfo} from "../lib/types";
+import { calculateVoterWeight } from "../lib/calculateVoterWeights";
+import {useQuery, UseQueryResult} from "@tanstack/react-query";
+
+type Args = UseVoterWeightPluginsArgs & {
+    plugins?: VoterWeightPluginInfo[],
+    tokenOwnerRecord?: ProgramAccount<TokenOwnerRecord>
+}
+
+// args, where all the necessary properties are marked as required - we can calculate once they are all set
+type RequiredArgs = Required<UseVoterWeightPluginsArgs> & {
+    plugins: VoterWeightPluginInfo[],
+    tokenOwnerRecord?: ProgramAccount<TokenOwnerRecord>
+};
+
+const argsAreSet = (args: Args): args is RequiredArgs =>
+    args.realmPublicKey !== undefined && args.governanceMintPublicKey !== undefined && args.walletPublicKey !== undefined &&
+    args.plugins !== undefined
+
+export const useCalculatedVoterWeight = (args: Args): UseQueryResult<CalculatedWeight> =>
+    useQuery(
```

Comment:
its a bit non-idiomatic to use useQuery for computed values; because you can get into situations where the input queries get invalidated, but this query fails to get invalidated. So I usually use `useAsync`. However this is imperfect too. sadly the react-query API is just not ideal for having a lot of nested queries. I think it's fine to leave it as a useQuery like you have it, but this is important to make note of.

Reply:
Good point. Might be a simple drop-in. I'll open a separate branch and experiment. If it results in performance issues I'll revert.